compiler/rustc_mir_transform/src/match_branches.rs

```diff
@@ -22,25 +24,68 @@ impl<'tcx> crate::MirPass<'tcx> for MatchBranchSimplification {
         let mut apply_patch = false;
         let mut patch = MirPatch::new(body);
         for (bb, bb_data) in body.basic_blocks.iter_enumerated() {
```

Comment:
Pre-existing: we examine terminators. Should we traverse the body in post-order?